### PR TITLE
Support views batch for panorama

### PR DESCRIPTION
**Pull request by Isotr0py**
- Support views batch in denosing loop for panorama pipeline

For P100 GPU, the speed was faster with `view_batch_size = 8` than `view_batch_size = 1` 

### test code
```python
import torch
from diffusers import StableDiffusionPanoramaPipeline, DPMSolverMultistepScheduler

view_batch_size = 8
seed = 86
model_ckpt = "stabilityai/stable-diffusion-2-base"
prompt = "a photo of the dolomites"

pipe = StableDiffusionPanoramaPipeline.from_pretrained(model_ckpt, torch_dtype=torch.float16).to("cuda")
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe.to("cuda")
generator = torch.Generator(device="cuda").manual_seed(seed)
image = pipe(prompt, generator=generator, num_inference_steps=25, width=2048, height=512, view_batch_size=view_batch_size).images[0]
display(image)

del pipe
torch.cuda.empty_cache()
```

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
cc @sayakpaul can you take a look?

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_panorama.py [hunk 1]
```diff
@@ -474,6 +474,7 @@ def __call__(
         width: Optional[int] = 2048,
         num_inference_steps: int = 50,
         guidance_scale: float = 7.5,
+        view_batch_size: int = 1,
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_panorama.py:477]: Let's also add an entry for this `arg` in the docstrings?

**Review by sayakpaul — COMMENTED** [review]
This looks good to me! Thanks so much! 

Some TODOs:

- [ ] Add an entry for the new argument. 
- [ ] Mention this argument in the [docs](https://github.com/huggingface/diffusers/blob/main/docs/source/en/api/pipelines/stable_diffusion/panorama.mdx) and comment on the potential speedup one might get. 
- [ ] Add a test to ensure robust behavior. 

Let me know if anything is unclear. 

Also cc: @omerbt

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_panorama.py [hunk 2]
```diff
@@ -631,42 +635,55 @@ def __call__(
...
                         t,
-                        encoder_hidden_states=prompt_embeds,
+                        encoder_hidden_states=prompt_embeds_input,
                         cross_attention_kwargs=cross_attention_kwargs,
                     ).sample
 
                     # perform guidance
                     if do_classifier_free_guidance:
-                        noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
+                        noise_pred_uncond, noise_pred_text = noise_pred[::2], noise_pred[1::2]
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_panorama.py:670]: Why do we have to do it this way?

Comment (Isotr0py) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_panorama.py:670]: For example, `noise_pred` with `view_batch_size=2` is `[uncond_view1, text_view1, uncond_view2, text_view2]`

If we use `noise_pred.chunk(2)`, `noise_pred_uncond, noise_pred_text = [uncond_view1, text_view1], [uncond_view2, text_view2]`, it's wrong splited.

The right output for guidance should be `noise_pred_uncond, noise_pred_text = [uncond_view1, uncond_view2], [text_view1, text_view2]`

**Review by sayakpaul — APPROVED** [review]
LGTM!

I think there's just one thing pending -- mention this argument from the doc and comment about the potential speedups.

**Comment by patrickvonplaten** [comment]
Good to merge for me whenever you want @sayakpaul

**Comment by omerbt** [comment]
Thanks for the PR @Isotr0py, looks good!

**Comment by sayakpaul** [comment]
Added a "tip" on `view_batch_size` in the doc. Will merge after the CI passes.

**Pull request merged**